tooling/metamodel-generator/src/test/java/org/hibernate/jpamodelgen/test/dao/Dao.java

```diff
@@ -47,6 +47,9 @@ public interface Dao {
     @HQL("where title like ?1")
     List<Book> findBooksByTitle(String title);
 
+    @HQL("where title like :book$title")
```

Comment:
I think `:book.title` is better although it's not consistent with method parameter name.

Reply:
Yes but if we're going to do that, then it has to be implemented in core, as proposed here: https://hibernate.atlassian.net/browse/HHH-17728

Which I think is probably a good idea, but one thing gives me pause: we used to have a feature similar to this (`Query.setBean()`) and few people ever seemed to use it, and Steve wanted to remove it in H6, which we did, and nobody appears to mind. 

So it would almost amount to putting _back_ in something we already removed. Now, that might still be justified because we have a new use for it, and we're proposing to do it in a slightly better way here. But it's not clear.